### PR TITLE
F ros control

### LINKED ISSUE #84: Running the driver without sudo
After a problem running the driver on the build server I have had a look at the Linux capabilities to run the launchfile as a normal user.

To run the ethercat driver we need:

```
sudo setcap cap_ipc_lock,cap_net_raw,cap_net_admin=eip /opt/ros/hydro/lib/pr2_ethercat/pr2_ethercat
```

To use the nice prefix we need:

```
sudo setcap cap_sys_nice=eip /usr/bin/nice
```

### LINKED ISSUE #120: Make sure all the libraries are exported
Add a `LIBRARIES` section exporting the libraries of the package to the catkin_package() macro, so that they can be found by other catkin packages.

We have done this for sr_ronex_transmissions in 5b217e5535d20a4c6d74f3bf1eafd02247fe5409

It should be checked on all the packages before closing this issue.

### LINKED ISSUE #128: Switch to ros_ethercat and ros_control

### LINKED ISSUE #136: transmission test fails
The solution for issue #104 caused the transmission test to fail. The transmission test propagates only once and checks the result. #105 ignores the first propagation to solve another problem.

Adolfo Rodríguez Tsouroukdissian (https://github.com/adolfo-rt)
has complained about this with email at 31 March

### PR DESCRIPTION
**Pull request by shadowmanos**
closed #128 
after merge git clone shadow_robot/ros_ethercat

closed #84 
Now just do "rosrun ros_ethercat_loop ros_grant" after catkin_make

closed #120
checked all CMakeLists.txt for as part of modifying for ros_control

closed #136
fixed the ronex transmission test in order to verify the ros_control modifications

### CODE REVIEW

File: sr_ronex_controllers/CMakeLists.txt [hunk 1]
```diff
@@ -5,26 +5,21 @@ cmake_minimum_required(VERSION 2.8.3)
...
 
-add_dependencies(sr_ronex_controllers sr_ronex_msgs_gencpp)
+add_dependencies(sr_ronex_controllers ${catkin_EXPORTED_TARGETS})
 
 # catkin_package parameters: http://ros.org/doc/groovy/api/catkin/html/dev_guide/generated_cmake_api.html#catkin-package
 catkin_package(
-    CATKIN_DEPENDS roscpp std_msgs std_srvs sr_ronex_msgs sr_ronex_hardware_interface pr2_mechanism_controllers pr2_controller_interface pr2_mechanism_model pluginlib
+    CATKIN_DEPENDS roscpp std_msgs std_srvs sr_ronex_msgs sr_ronex_hardware_interface controller_interface ros_ethercat_model pluginlib
     INCLUDE_DIRS include
+    LIBRARIES sr_ronex_controllers
```
Comment (shadowmanos) [inline sr_ronex_controllers/CMakeLists.txt:22]: for #120

File: sr_ronex_controllers/scripts/load_passthrough_controllers.py [hunk 2]
```diff
@@ -17,11 +17,10 @@
 # License along with this library.
 # ####################################################################
 
-import roslib; roslib.load_manifest('sr_ronex_controllers')
```
Comment (shadowmanos) [inline sr_ronex_controllers/scripts/load_passthrough_controllers.py:20]: no needed in catkin/hydro

File: sr_ronex_controllers/src/general_io_passthrough_controller.cpp [hunk 3]
```diff
@@ -33,19 +33,7 @@ namespace ronex
...
-  {
-    for(size_t i=0; i < digital_subscribers_.size(); ++i)
-    {
-      digital_subscribers_[i].shutdown();
-    }
-    for(size_t i=0; i < pwm_subscribers_.size(); ++i)
-    {
-      pwm_subscribers_[i].shutdown();
-    }
-  }
```
Comment (shadowmanos) [inline sr_ronex_controllers/src/general_io_passthrough_controller.cpp:46]: done automatically when subscribers go out of scope

File: sr_ronex_drivers/CMakeLists.txt [hunk 4]
```diff
@@ -1,22 +1,14 @@
 # Catkin User Guide: http://www.ros.org/doc/groovy/api/catkin/html/user_guide/user_guide.html
 # Catkin CMake Standard: http://www.ros.org/doc/groovy/api/catkin/html/user_guide/standards.html
 
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fprofile-arcs -ftest-coverage")
```
Comment (shadowmanos) [inline sr_ronex_drivers/CMakeLists.txt:4]: apparently forgotten from some early coverage tests

File: sr_ronex_drivers/CMakeLists.txt [hunk 5]
```diff
@@ -25,21 +17,23 @@ add_library(
...
 
-add_dependencies(sr_ronex_drivers sr_ronex_msgs_gencpp ${PROJECT_NAME}_gencfg)
+add_dependencies(sr_ronex_drivers ${catkin_EXPORTED_TARGETS} ${PROJECT_NAME}_gencfg)
 
 # catkin_package parameters: http://ros.org/doc/groovy/api/catkin/html/dev_guide/generated_cmake_api.html#catkin-package
 catkin_package(
-    CATKIN_DEPENDS pr2_msgs ethercat_hardware std_msgs sr_ronex_msgs pr2_hardware_interface eml roscpp rostest realtime_tools diagnostic_msgs diagnostic_updater pluginlib sr_ronex_external_protocol sr_ronex_hardware_interface dynamic_reconfigure tinyxml
+    CATKIN_DEPENDS ros_ethercat_hardware std_msgs sr_ronex_msgs ros_ethercat_model roscpp rostest realtime_tools diagnostic_msgs diagnostic_updater pluginlib sr_ronex_external_protocol sr_ronex_hardware_interface dynamic_reconfigure
     INCLUDE_DIRS include
+    LIBRARIES sr_ronex_drivers
```
Comment (shadowmanos) [inline sr_ronex_drivers/CMakeLists.txt:28]: for #120

File: sr_ronex_drivers/ethercat_device_plugin.xml [hunk 6]
```diff
@@ -5,32 +5,32 @@
...
   <class name="sr_ronex_drivers/7" type="StandardEthercatDevice" base_class_type="EthercatDevice">
     <description>
-      RoNeX bridge - duplicated for backward compatibility issues.
+      RoNeX - Bridge module
     </description>
   </class>
   <class name="sr_ronex_drivers/73946162" type="StandardEthercatDevice" base_class_type="EthercatDevice">
     <description>
-      Beckhoff CU1128 bridge - Product ID is 73946162 (0x4685432).
+      RoNeX - Beckhoff CU1128 bridge module
```
Comment (shadowmanos) [inline sr_ronex_drivers/ethercat_device_plugin.xml]: The description gets printed on screen now by ros_ethercat as a device name. Deleted the ugly hex numbers.

File: sr_ronex_drivers/include/sr_ronex_drivers/cod_decod/cod_decod_std_io.hpp [hunk 7]
```diff
@@ -226,12 +225,12 @@ namespace sr_cod_decod
...
-    realtime_tools::RealtimePublisher<std_msgs::UInt16MultiArray> *analog_input_state_publisher_;
+    boost::scoped_ptr<realtime_tools::RealtimePublisher<sr_ronex_msgs::BoolArray> > digital_input_state_publisher_;
+    boost::scoped_ptr<realtime_tools::RealtimePublisher<std_msgs::UInt16MultiArray> >analog_input_state_publisher_;
 
-    realtime_tools::RealtimeBox<boost::shared_ptr<const sr_ronex_msgs::BoolArray> > digital_output_;
-    realtime_tools::RealtimeBox<boost::shared_ptr<const std_msgs::UInt16MultiArray> > analog_output_;
-    realtime_tools::RealtimeBox<boost::shared_ptr<const std_msgs::UInt16MultiArray> > PWM_output_;
+    realtime_tools::RealtimeBox<boost::shared_ptr<sr_ronex_msgs::BoolArray> > digital_output_;
+    realtime_tools::RealtimeBox<boost::shared_ptr<std_msgs::UInt16MultiArray> > analog_output_;
+    realtime_tools::RealtimeBox<boost::shared_ptr<std_msgs::UInt16MultiArray> > PWM_output_;
```
Comment (shadowmanos) [inline sr_ronex_drivers/include/sr_ronex_drivers/cod_decod/cod_decod_std_io.hpp:233]: The compiler and Eclipse complained about the const here. Not sure what they started complaining now.

File: sr_ronex_drivers/src/cod_decod/cod_decod_std_io.cpp [hunk 8]
```diff
@@ -125,27 +125,18 @@ namespace sr_cod_decod
...
   }
 
-  CodDecodStdIo::~CodDecodStdIo()
-  {
-    if (digital_input_state_publisher_) delete digital_input_state_publisher_;
-    if (analog_input_state_publisher_) delete analog_input_state_publisher_;
-    sub_digital_output_command_.shutdown();
-    sub_analog_output_command_.shutdown();
-    sub_PWM_output_command_.shutdown();
-  }
```
Comment (shadowmanos) [inline sr_ronex_drivers/src/cod_decod/cod_decod_std_io.cpp:147]: digital_input_state_publisher_  and  analog_input_state_publisher_ are now smart pointers.
shutdown is done automatically by the framework

File: sr_ronex_drivers/src/sr_board_0x.cpp [hunk 9]
```diff
@@ -142,24 +131,6 @@ void SrBoard0X::construct(EtherCAT_SlaveHandler *sh, int &start_address)
...
-{
-  return EthercatDevice::readData(com, address, data, length, FIXED_ADDR);
-}
-
-
-int SrBoard0X::writeData(EthercatCom *com, EC_UINT address, void const *data, EC_UINT length)
-{
-  return EthercatDevice::writeData(com, sh_, address, data, length, FIXED_ADDR);
-}
-
```
Comment (shadowmanos) [inline sr_ronex_drivers/src/sr_board_0x.cpp:162]: the compiler does all the above automatically

File: sr_ronex_drivers/src/sr_board_mk2_gio.cpp [hunk 10]
```diff
@@ -38,21 +40,29 @@
...
+  for (size_t i = 0; i < general_io_->command_.digital_.size(); ++i)
+    ros::param::del(general_io_name + "input_mode_" + lexical_cast<string>(i));
+  for (size_t i = 0; i < general_io_->command_.pwm_.size(); ++i)
+    ros::param::del(general_io_name + "pwm_period_" + lexical_cast<string>(i));
 
-  delete sh_->get_fmmu_config();
-  delete sh_->get_pd_config();
+  string controller_name = "/ronex_" + serial_number_ + "_passthrough/";
+  ros::param::del(controller_name + "ronex_id");
+  ros::param::del(controller_name + "type");
```
Comment (shadowmanos) [inline sr_ronex_drivers/src/sr_board_mk2_gio.cpp:61]: the deleting is now done in ros_ethercat.
all the parameters related to this device are now deleted from the server. Something like that may cause #114 so it should be replicated in F_spi_module.

File: sr_ronex_drivers/src/sr_board_mk2_gio.cpp [hunk 11]
```diff
@@ -159,12 +167,10 @@ void SrBoardMk2GIO::construct(EtherCAT_SlaveHandler *sh, int &start_address)
 
   sh->set_pd_config(pd);
 
-  ROS_INFO("status_size_ : %d ; command_size_ : %d", status_size_, command_size_);
-
```
Comment (shadowmanos) [inline sr_ronex_drivers/src/sr_board_mk2_gio.cpp:163]: this info is printed in the previous ROS_INFO line

File: sr_ronex_drivers/src/sr_board_mk2_gio.cpp [hunk 12]
```diff
@@ -75,8 +85,6 @@ void SrBoardMk2GIO::construct(EtherCAT_SlaveHandler *sh, int &start_address)
   device_name_ = ronex::build_name( product_alias_, ronex_id_ );
 
   EthercatDevice::construct(sh,start_address);
-  sh->set_fmmu_config( new EtherCAT_FMMU_Config(0) );
-  sh->set_pd_config( new EtherCAT_PD_Config(0) );
```
Comment (shadowmanos) [inline sr_ronex_drivers/src/sr_board_mk2_gio.cpp:79]: ros_ethercat takes care of this now

File: sr_ronex_drivers/src/sr_tcat.cpp [hunk 13]
```diff
@@ -49,9 +49,6 @@ SrTCAT::~SrTCAT()
   std::stringstream param_path;
   param_path << "/ronex/devices/" << parameter_id_ ;
   ros::param::del(param_path.str());
-
-  delete sh_->get_fmmu_config();
-  delete sh_->get_pd_config();
```
Comment (shadowmanos) [inline sr_ronex_drivers/src/sr_tcat.cpp:54]: ros_ethercat take care of this now

File: sr_ronex_drivers/src/sr_board_mk2_gio.cpp [hunk 14]
```diff
@@ -177,30 +183,18 @@ int SrBoardMk2GIO::initialize(pr2_hardware_interface::HardwareInterface *hw, boo
...
-{
-  return EthercatDevice::readData(com, address, data, length, FIXED_ADDR);
-}
-
-
-int SrBoardMk2GIO::writeData(EthercatCom *com, EC_UINT address, void const *data, EC_UINT length)
-{
-  return EthercatDevice::writeData(com, sh_, address, data, length, FIXED_ADDR);
-}
-
```
Comment (shadowmanos) [inline sr_ronex_drivers/src/sr_board_mk2_gio.cpp:203]: compiler does that automatically

File: sr_ronex_drivers/src/sr_tcat.cpp [hunk 15]
```diff
@@ -74,8 +71,6 @@ void SrTCAT::construct(EtherCAT_SlaveHandler *sh, int &start_address)
   device_name_ = ronex::build_name( product_alias_, ronex_id_ );
 
   EthercatDevice::construct(sh,start_address);
-  sh->set_fmmu_config( new EtherCAT_FMMU_Config(0) );
-  sh->set_pd_config( new EtherCAT_PD_Config(0) );
```
Comment (shadowmanos) [inline sr_ronex_drivers/src/sr_tcat.cpp:78]: ros_ethercat take care of this now

File: sr_ronex_drivers/src/standard_ethercat_device.cpp [hunk 16]
```diff
@@ -38,24 +38,7 @@
...
-  delete sh_->get_fmmu_config();
-  delete sh_->get_pd_config();
-}
-
-void StandardEthercatDevice::construct(EtherCAT_SlaveHandler *sh, int &start_address)
-{
-  EthercatDevice::construct(sh,start_address);
-  sh->set_fmmu_config( new EtherCAT_FMMU_Config(0) );
-  sh->set_pd_config( new EtherCAT_PD_Config(0) );
-}
```
Comment (shadowmanos) [inline sr_ronex_drivers/src/standard_ethercat_device.cpp:56]: ros_ethercat take care of this now

File: sr_ronex_drivers/test/test_ethercat_drivers.cpp [hunk 17]
```diff
@@ -55,9 +57,15 @@ TEST(RonexUtils, constructor )
...
 
-  pr2_hardware_interface::HardwareInterface hw;
-  int retsbm = sbm.initialize( &hw );
+  ros::NodeHandle nh;
+  string xml_string;
+  nh.getParam("robot_description", xml_string);
+  TiXmlDocument urdf_xml;
+  urdf_xml.Parse(xml_string.c_str());
+  TiXmlElement *root = urdf_xml.FirstChildElement("robot");
+  ros_ethercat_model::RobotState hw(root);
```
Comment (shadowmanos) [inline sr_ronex_drivers/test/test_ethercat_drivers.cpp:66]: ros_ethercat_model::RobotState needs a xacro to start and initialize transmissions and GeneralIO

File: sr_ronex_drivers/test/test_ethercat_drivers.test [hunk 18]
```diff
@@ -1,3 +1,4 @@
 <launch>
+  <param name="robot_description" command="$(find xacro)/xacro.py '$(find sr_ronex_examples)/model/simple_robot.urdf'" />
```
Comment (shadowmanos) [inline sr_ronex_drivers/test/test_ethercat_drivers.test:2]: ros_ethercat_model::RobotState needs a xacro to start and initialize transmissions and GeneralIO

File: sr_ronex_examples/CMakeLists.txt [hunk 19]
```diff
@@ -26,8 +26,9 @@ include_directories(include ${catkin_INCLUDE_DIRS})
 # You must invoke catkin_package() before adding any targets (libraries and executables).
 # The reason is because catkin_package() will change the location where the targets are built.
 catkin_package(
-  CATKIN_DEPENDS roscpp rospy std_msgs std_srvs sr_ronex_msgs sr_ronex_utilities dynamic_reconfigure sr_ronex_hardware_interface pr2_mechanism_controllers pr2_controller_interface pr2_mechanism_model pluginlib
+  CATKIN_DEPENDS roscpp rospy std_msgs std_srvs sr_ronex_msgs sr_ronex_utilities dynamic_reconfigure sr_ronex_hardware_interface realtime_tools ros_ethercat_model pluginlib controller_interface
   INCLUDE_DIRS include
+  LIBRARIES sr_ronex_simple_controller
```
Comment (shadowmanos) [inline sr_ronex_examples/CMakeLists.txt:31]: for #120

File: sr_ronex_launch/CMakeLists.txt [hunk 20]
```diff
@@ -15,7 +15,7 @@ set(LIBRARY_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/lib)
 
 # catkin_package parameters: http://ros.org/doc/groovy/api/catkin/html/dev_guide/generated_cmake_api.html#catkin-package
 catkin_package(
-    CATKIN_DEPENDS pr2_ethercat
+    CATKIN_DEPENDS
```
Comment (shadowmanos) [inline sr_ronex_launch/CMakeLists.txt:18]: pr2_ethercat is replaced by ros_ethercat and they are both runtime dependencies only

File: sr_ronex_launch/package.xml [hunk 21]
```diff
@@ -18,10 +18,9 @@
   <buildtool_depend>catkin</buildtool_depend>
 
   <!-- Dependencies needed to compile this package. -->
-  <build_depend>pr2_ethercat</build_depend>
 
   <!-- Dependencies needed after this package is compiled. -->
-  <run_depend>pr2_ethercat</run_depend>
+  <run_depend>ros_ethercat_loop</run_depend>
```
Comment (shadowmanos) [inline sr_ronex_launch/package.xml:23]: pr2_ethercat is replaced by ros_ethercat and they are both runtime dependencies only

File: sr_ronex_transmissions/test/test_ronex_transmission.cpp [hunk 22]
```diff
@@ -78,19 +76,13 @@ TEST(RonexTransmission, propagateCommand)
...
-  pr2_mechanism_model::RobotState state(&model);
-
   //setting effort for the joint
-  state.joint_states_[0].commanded_effort_ = 5.1;
+  state.joint_states_["joint1"].commanded_effort_ = 5.1;
 
   general_io->command_.pwm_[1].period =  64000;  //setting period too
 
   state.propagateJointEffortToActuatorEffort();
+  state.propagateJointEffortToActuatorEffort();
```
Comment (shadowmanos) [inline sr_ronex_transmissions/test/test_ronex_transmission.cpp:85]: the second propagation fixes #136

File: sr_ronex_controllers/include/sr_ronex_controllers/fake_calibration_controller.hpp [hunk 23]
```diff
@@ -28,30 +28,29 @@
...
 
 namespace ronex
 {
   class FakeCalibrationController
-    : public pr2_controller_interface::Controller
+    : public controller_interface::Controller<ros_ethercat_model::RobotState>
   {
   public:
     FakeCalibrationController();
-    virtual ~FakeCalibrationController();
```
Comment (ugocupcic) [inline sr_ronex_controllers/include/sr_ronex_controllers/fake_calibration_controller.hpp:44]: why delete the destructor?

File: sr_ronex_controllers/scripts/load_passthrough_controllers.py [hunk 24]
```diff
@@ -86,13 +85,13 @@ def load_and_start_ctrl(self, ronex_ids):
         rospy.loginfo("Starting controllers: " + str(controllers_list))
 
         # calling the services to load and switch the controllers on
-        rospy.wait_for_service('pr2_controller_manager/list_controllers')
-        rospy.wait_for_service('pr2_controller_manager/load_controller')
-        rospy.wait_for_service('pr2_controller_manager/switch_controller')
```
Comment (ugocupcic) [inline sr_ronex_controllers/scripts/load_passthrough_controllers.py:91]: might be good to have a 

```
loginfo("Waiting for services: list/load/switch")`
rospy.wait_for_service...
loginfo("Services found')
```

Thoughts?

File: sr_ronex_drivers/ethercat_device_plugin.xml [hunk 25]
```diff
@@ -5,32 +5,32 @@
   -->
   <class name="sr_ronex_drivers/87032868" type="SrBoard0X" base_class_type="EthercatDevice">
     <description>
-      SrBoard0X - Shadow Robot RoNeX Board
+      RoNeX - General I/O module
```
Comment (ugocupcic) [inline sr_ronex_drivers/ethercat_device_plugin.xml]: is that mk1 ronex? May be add (compatibility version) or something? it's weird to have twice the same description in the file

File: sr_ronex_drivers/ethercat_device_plugin.xml [hunk 26]
```diff
@@ -5,32 +5,32 @@
...
   </class>
   <class name="sr_ronex_drivers/33554432" type="StandardEthercatDevice" base_class_type="EthercatDevice">
     <description>
-      RoNeX bridge - Product ID is 0x02000000.
+      RoNeX - Bridge module
     </description>
   </class>
   <class name="sr_ronex_drivers/7" type="StandardEthercatDevice" base_class_type="EthercatDevice">
     <description>
-      RoNeX bridge - duplicated for backward compatibility issues.
```
Comment (ugocupcic) [inline sr_ronex_drivers/ethercat_device_plugin.xml:28]: same as above. 

I don't mind if you feel against it.

File: sr_ronex_drivers/include/sr_ronex_drivers/sr_board_0x.h [hunk 27]
```diff
@@ -38,17 +38,11 @@ class SrBoard0X : public StandardEthercatDevice
 {
 public:
   virtual void construct(EtherCAT_SlaveHandler *sh, int &start_address);
-  virtual int initialize(pr2_hardware_interface::HardwareInterface *hw, bool allow_unprogrammed=true);
 
-  SrBoard0X();
```
Comment (ugocupcic) [inline sr_ronex_drivers/include/sr_ronex_drivers/sr_board_0x.h:43]: why delete ?

File: sr_ronex_drivers/src/sr_board_mk2_gio.cpp [hunk 28]
```diff
@@ -38,21 +36,29 @@
 PLUGINLIB_EXPORT_CLASS(SrBoardMk2GIO, EthercatDevice);
 
 const std::string SrBoardMk2GIO::product_alias_ = "general_io";
+using boost::lexical_cast;
+
 
 SrBoardMk2GIO::SrBoardMk2GIO() :
-  EthercatDevice(), node_("~"), cycle_count_(0), has_stacker_(false)
```
Comment (ugocupcic) [inline sr_ronex_drivers/src/sr_board_mk2_gio.cpp:43]: why not calling base class constructor?

File: sr_ronex_examples/src/sr_ronex_simple_controller.cpp [hunk 29]
```diff
@@ -37,12 +37,12 @@ SrRoNeXSimpleController::~SrRoNeXSimpleController()
...
 }
 
-bool SrRoNeXSimpleController::init(pr2_mechanism_model::RobotState* robot, ros::NodeHandle &n)
+bool SrRoNeXSimpleController::init(ros_ethercat_model::RobotState* robot, ros::NodeHandle &n)
 {
   assert (robot);
   
   std::string path("/ronex/general_io/test_ronex");
-  general_io_ = static_cast<ronex::GeneralIO*>( robot->model_->hw_->getCustomHW(path) );
+  general_io_ = static_cast<ronex::GeneralIO*>( robot->getCustomHW(path) );
```
Comment (ugocupcic) [inline sr_ronex_examples/src/sr_ronex_simple_controller.cpp:45]: Probably means updating the wiki at some point. Can you add a ticket to reflect that?

Thinking about [this one](https://github.com/shadow-robot/sr-ronex/wiki/Accessing-the-ronex-data-directly-from-a-controller) from the top of my head but there could be others.

File: sr_ronex_controllers/include/sr_ronex_controllers/fake_calibration_controller.hpp (hunk 23 above)
Comment (shadowmanos) [inline sr_ronex_controllers/include/sr_ronex_controllers/fake_calibration_controller.hpp:44]: complier will define a default empty destructor and inline it. I am only aware of 3 cases where an empty destructor absolutely has to be defined.
- virtual destructor in base class (not in any derived) (Eclipse can spot that)
- class has a member that is incomplete (forward declaration) (compiler error)
- class defined in Windows .dll that uses different C++ runtime than .exe (hours of fun!)

If you absolutely want to see an empty destructor I will (re)define it albeit inline.

File: sr_ronex_examples/src/sr_ronex_simple_controller.cpp (hunk 29 above)
Comment (shadowmanos) [inline sr_ronex_examples/src/sr_ronex_simple_controller.cpp:45]: added ticket #146

File: sr_ronex_drivers/src/sr_board_mk2_gio.cpp (hunk 28 above)
Comment (shadowmanos) [inline sr_ronex_drivers/src/sr_board_mk2_gio.cpp:43]: default constructor of base class will get called anyway. same with destructor.

File: sr_ronex_drivers/include/sr_ronex_drivers/sr_board_0x.h (hunk 27 above)
Comment (shadowmanos) [inline sr_ronex_drivers/include/sr_ronex_drivers/sr_board_0x.h:43]: empty default constructors don't need to be defined. The compiler provides an inline one and constructs the same things with the user supplied one.

File: sr_ronex_drivers/ethercat_device_plugin.xml (hunk 25 above)
Comment (shadowmanos) [inline sr_ronex_drivers/ethercat_device_plugin.xml]: had quick chat with Robert and renamed according to recommendation. SrBoard0X is actually mk2 he says, and SrBoardMk2GIO refers to mk3

File: sr_ronex_controllers/scripts/load_passthrough_controllers.py (hunk 24 above)
Comment (shadowmanos) [inline sr_ronex_controllers/scripts/load_passthrough_controllers.py:91]: I don't think enough time is spent waiting here to require a message. I can add some messages for the actual loading and starting of controllers.

Comment (ugocupcic) [inline sr_ronex_controllers/scripts/load_passthrough_controllers.py:91]: No, my thought was if the services aren't available then it'll hang there waiting for them.

File: sr_ronex_controllers/include/sr_ronex_controllers/fake_calibration_controller.hpp (hunk 23 above)
Comment (ugocupcic) [inline sr_ronex_controllers/include/sr_ronex_controllers/fake_calibration_controller.hpp:44]: Nope I just didn't know that!

**Pull request merged**